api/app/tests/weather_models/test_models_predictions_db.py

```diff
@@ -65,47 +70,40 @@ def mock_get_model_run_predictions(*args):
                             mock_get_model_run_predictions)
 
 
-@ scenario("test_models_predictions_db.feature", "Get model predictions from database")
+@pytest.mark.usefixtures("mock_jwt_decode")
+@scenario("test_models_predictions_db.feature", "Get model predictions from database",
+          example_converters=dict(codes=json.loads,
+                                  endpoint=str, data=str,
+                                  num_prediction_values=json.loads,
+                                  expected_response=load_json_file))
 def test_db_predictions_scenario():
     """ BDD Scenario for predictions """
 
 
-@given("A database with <data>")
-def given_a_database(mock_session, data):  # pylint: disable=redefined-outer-name,unused-argument
+@ given("A database with <data>", target_fixture='context')
+def given_a_database(data):
     """ Bind the data variable """
+    assert data
     return {}
 
 
-@given("station <codes>")
-def given_stations(codes):
-    """ Turn provided string into array of codes. """
-    return eval(codes)  # pylint: disable=eval-used
-
-
-# pylint: disable=unused-argument, redefined-outer-name
-@when("I call <endpoint>")
-def when_predictions(mock_jwt_decode, given_a_database, given_stations, endpoint: str):
+@ when("I call <endpoint> with <codes>")
+def when_predictions(context, endpoint: str, codes: List):
     """ post to endpoint """
     client = TestClient(app.main.app)
     response = client.post(
-        endpoint, headers={'Authorization': 'Bearer token'}, json={'stations': given_stations})
-    given_a_database['response_json'] = response.json()
-# pylint: enable=unused-argument, redefined-outer-name
+        endpoint, headers={'Authorization': 'Bearer token'}, json={'stations': codes})
+    context['response_json'] = response.json()
 
 
-@then('There are <num_prediction_values>')
-def assert_num_predictions(given_a_database, num_prediction_values):  # pylint: disable=redefined-outer-name
+@ then('There are <num_prediction_values>')
+def assert_num_predictions(context, num_prediction_values):
     """ Even though there are only two predictions in the database, we expect an interpolated noon value. """
-    num_prediction_values = eval(num_prediction_values)  # pylint: disable=eval-used
-    assert len(given_a_database['response_json']['predictions']
-               [num_prediction_values['index']]['values']) == num_prediction_values['len']
+    assert len(context['response_json']['predictions'][num_prediction_values['index']]
+               ['values']) == num_prediction_values['len']
 
 
-@then('The <expected_response> is matched')
-def assert_response(given_a_database, expected_response):  # pylint: disable=redefined-outer-name
+@ then('The <expected_response> is matched')
```

Comment:
but then you added spaces here. So 🤷‍♀️. 
I'm not actually bothered by it either way

Reply:
I'm VERY bothered! :) Thanks.